Automattic/jetpack
# Jetpack Sync: Switch 'jetpack_connection_active_plugins' to a callable

**Do not merge till the WPCOM counterpart diff is merged**: D153988-code

Start syncing `jetpack_connection_active_plugins` as a callable.
This approach will address edge cases where the site option would not pick up the change therefore the corresponding option on WPCOM would end up out-of-sync.
Since callables are checked every time we send remote Sync actions to WPCOM, we are expecting that switching to a callable will ensure the option is more aligned.
Checkout the linked product discussion below for full context and research.

## Proposed changes:

* `Automattic\Jetpack\Sync\Defaults`: Add `jetpack_connection_active_plugins` to callable whitelist and remove from options whitelist
* `Automattic\Jetpack\Sync\Data_Settings`: Add `jetpack_connection_active_plugins` to the callables we always sync, no matter the consumer configuration and remove from the options we always sync.
* `Automattic\Jetpack\Sync\Functions`: Introduce `get_jetpack_connection_active_plugins` method for fetching the active plugins that are using the Jetpack Connection

### Other information:


## Jetpack product discussion


p9dueE-8gv-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






**Important**: make sure to test with the WPCOM counterpart diff applied: D153988-code

We want to test various combinations of standalone Jetpack products and activation/deactivation flows:
* Jetpack plugin, Backups, Protect, WooCommerce, WooCommerce Payments etc
* Activate/deactivate various plugins in the list above and ensure the option is up-to-date on WPCOM
* Make sure to test with a case where the first/last active plugin is WooCommerce (uses only the Jetpack Connection and not Jetpack Sync) and ensure the option is still up-to-date on WPCOM.
* Test deactivating/activating plugins via `cli` with/without `--skip-plugins`. When activating/deactivating using the `--skip-plugins` option, make sure the callable will get sent to WPCOM with the next batch of Sync actions.
